### DISCUSSION TITLE
Can Renovate be run on an event-driven basis?

### DISCUSSION THREAD

**Discussion by ZackKanter** [renovatebot/renovate#22574] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Rather than running on a cron job, would like to trigger Renovate to run in response to an event in order to achieve [something like this](https://aws.amazon.com/blogs/devops/keeping-up-with-your-dependencies-building-a-feedback-loop-for-shared-libraries/).

### What you tried so far.

Reading the docs, but haven't found anything so far.

**Comment by rarkins** [comment, marked as answer]
Not natively with Renovate the CLI, but the hosted app works somewhat like this (responding and running according to events of importance in the repo, such as changing of config, package files, or PRs).

**Reply by ZackKanter** [reply]
Thank you! So, the workaround would be basically instead of [pointing cron at a script](https://github.com/renovatebot/renovate/blob/a700ccc4b8a3f67e0cfa349f1530e36f6054aa1e/docs/usage/examples/self-hosting.md#usage) to kick off Renovate, you would point an event at the script?

> Most people use cron to schedule when Renovate runs, usually on an hourly schedule. Here's an example Bash script that you can point cron to

**Reply by rarkins** [reply]
That would likely to be too simple, because you may have multiple - even dozens - of events in quick succession and you don't want to run too many times on the same repo.

**Reply by viceice** [reply]
you can probably use the free mend renovate on premise, which has support for cron and event driven runs
